lib/sparkql/function_resolver.rb

```diff
@@ -118,6 +122,13 @@ def call()
   protected 
   
   # Supported function calls
+
+  def regex(regular_expression)
+    {
+      :type => :character,
+      :value => regular_expression
```

Comment:
Good call on validating. I'm indifferent to returning a regex since we won't typically use it as a regex within ruby.